warehouse/upload.go

```diff
@@ -1190,6 +1190,12 @@ func (job *UploadJob) loadUserTables(loadFilesTableMap map[tableNameT]bool) ([]e
 			return job.processLoadTableResponse(map[string]error{job.usersTableName(): err})
 		}
 	}
+
+	// Skip loading user tables if identifies table schema is not present
+	if identifiesSchema := job.GetTableSchemaInUpload(warehouseutils.IdentifiesTable); len(identifiesSchema) == 0 {
+		return []error{}, nil
+	}
+
```

Comment:
Instead of keeping [this](https://github.com/rudderlabs/rudder-server/blob/ea3bbd502d7ed5f6f9be0e08e1a9af389589831a/warehouse/integrations/snowflake/snowflake.go#L552) check at the integration level, moving this out.